pkg/route/validation/validation.go

```diff
@@ -342,6 +370,30 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaderResponse    = []string{"strict-transport-security"}
+	notAllowedHTTPHeaderResponseSet = sets.NewString(notAllowedHTTPHeaderResponse...)
+)
+
+func validateHeaders(path *field.Path, headerValues []routev1.RouteHTTPHeader) *field.Error {
+	for key, value := range headerValues {
+		if value.Action.Set != nil && len(value.Action.Set.Value) != 0 && len(value.Action.Set.Value) > 4096 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("value exceeds the maximum length which is 2048"))
+		}
+		if value.Action.Set != nil && len(value.Action.Set.Value) == 0 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("minimum length of value shall be one."))
+		}
+		if notAllowedHTTPHeaderResponseSet.Has(strings.ToLower(value.Name)) {
+			return field.NotSupported(path.Index(key).Child("name"), value.Name, []string{fmt.Sprintf("All headers except %v", strings.Join(notAllowedHTTPHeaderResponse, ","))})
+		}
+		r, _ := regexp.Compile("^(?:%(?:%|(?:\\{[-+]?[QXE](,[-+]?[QXE])*\\})?\\[(?:req\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]])*$")
```

Comment:
I don't think we should ignore the error from `.Compile()`. Although there's no error in the current regexp let's make it fail fast for future amendments. 

Make it a global, ala:


```go
var headerRe = regexp.MustCompile("^(?:%(?:%|(?:\\{[-+]?[QXE](,[-+]?[QXE])*\\})?\\[(?:req\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]])*$")
```

Reply:
If you use a string literal you can drop the double backslashes. 

```go
var headerRe = regexp.MustCompile(`^(?:%(?:%|(?:\{[-+]?[QXE](,[-+]?[QXE])*\})?\[(?:req\.hdr\([0-9A-Za-z-]+\)|ssl_c_der)(?:,(?:lower|base64))*\])|[^%[:cntrl:]])*$`)
```

Then you can plumb that string literal into https://regex101.com/ to interactively test without _un-escaping_ the string.